File: johannes-larsson-ruby/day2/part2.rb
def compareWords(word1, word2)
	i = 0
	index = -1
	until i >= word1.length do
		if (word1[i] != word2[i]) then
			if (index != -1) then
				return false
			else
				index = i
			end
		end
		i += 1
	end
	
	return word1[0..index-1] + word1[index+1..word1.length-1]
end


words = []

File.open('input').each do |line|
	words.each { |word|
		res = compareWords(word,line)
		if (res != false)
			puts res
			exit
		end
	}
	words.push(line)
end

File: sgenheden-python/day2/part2.py
import fileinput


def compare_words(word1, word2):
    idx = -1
    nequal = 0
    for i, (l1, l2) in enumerate(zip(word1, word2)):
        if l1 != l2:
            idx = i
        else:
            nequal += 1
    if nequal == len(word1)-1:
        return word1[:idx]+word1[idx+1:]


if __name__ == "__main__":
    inp = [line.strip() for line in fileinput.input()]
    ret = None
    for i, word1 in enumerate(inp):
        for word2 in inp[i+1:]:
            ret = compare_words(word1, word2)
            if ret is not None:
                print(f"Commong word {ret}")
                break
        if ret is not None:
            break
